lib/tasks/themes.rake

```diff
@@ -292,3 +292,96 @@ task "themes:qunit_all_official" => :environment do |task, args|
   Rake::Task["themes:qunit"].reenable
   Rake::Task["themes:qunit"].invoke("ids", theme_ids_with_qunit_tests.join("|"))
 end
+
+HORIZON_REMOTE_PATHS = %w[
+  github.com/discourse-org/horizon
+  github.com/discourse-org/nextgen-theme
+  github.com/discourse/horizon
+]
+
+def get_horizon_themes
+  Theme.joins(:remote_theme).where(<<~SQL)
+    remote_themes.remote_url LIKE ANY (ARRAY[#{HORIZON_REMOTE_PATHS.map { |path| "'%#{path}%'" }.join(", ")}])
+  SQL
+end
+
+desc "Remove duplicated Horizon themes"
+task "themes:deduplicate_horizon" => :environment do |task, args|
+  horizon_themes = get_horizon_themes
+
+  abort "🟩 No Horizon themes found" if horizon_themes.empty?
+
+  # v0 will handle only one horizon theme
+  abort("⭕ More than one horizon theme installed") if horizon_themes.count > 1
+
+  manual_horizon_theme = horizon_themes.first
+  system_horizon_theme = Theme.horizon_theme
+
+  Theme.transaction do
+    # Enable hidden horizon theme if it is not enabled
+    if !SiteSetting.experimental_system_themes_map.include?("horizon")
+      SiteSetting.experimental_system_themes =
+        (SiteSetting.experimental_system_themes_map << "horizon").join("|")
+    end
+
+    # Make the system horizon selectable if manual horizon is selectable
+    system_horizon_theme.update!(user_selectable: manual_horizon_theme.user_selectable)
+
+    # Move all components of the manual horizon theme to the system horizon theme
+    ChildTheme
+      .where(parent_theme_id: manual_horizon_theme.id)
+      .where.not(child_theme_id: system_horizon_theme.child_themes.pluck(:id))
+      .find_each { |child_theme| child_theme.update!(parent_theme_id: system_horizon_theme.id) }
```

Comment:
```suggestion
      .update_all!(parent_theme_id: system_horizon_theme.id)
```

I think you can just do this here? Unless you specifically want to run the validations for `update!` on each one